jsonpb/jsonpb_test.go

```diff
@@ -789,6 +789,18 @@ func TestUnmarshalJSONPBUnmarshaler(t *testing.T) {
 	if msg.rawJson != rawJson {
 		t.Errorf("message contents not set correctly after unmarshalling JSON: got %s, wanted %s", msg.rawJson, rawJson)
 	}
+
+	rawJson = `{"stringField":null}`
+	var ptrFieldMsg ptrFieldMessage
+	if err := Unmarshal(strings.NewReader(rawJson), &ptrFieldMsg); err != nil {
+		t.Errorf("an unexpected error occurred when parsing into JSONPBUnmarshaler: %v", err)
+	}
+	if ptrFieldMsg.StringField == nil {
+		t.Errorf("ptr field which implements JSONPBUnmarshaler is nil after unmarshalling with JSON {\"stringField\":null}:")
+	}
+	if ptrFieldMsg.StringField != nil && ptrFieldMsg.StringField.IsSet == false {
```

Comment:
Not sure if you need the IsSet field.  But if you really want to keep it, the validation here should not need to include check for StringField != nil, i.e. it should be independent.

Reply:
Removed this validation, "proto.Equal" is enough, but "IsSet" is not removed, I'd like to use it to make sure `func (s *stringField) UnmarshalJSONPB` is executed.